kms-provider-kroxylicious-inmemory/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryEdekSerde.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.nio.ByteBuffer;
+
+import io.kroxylicious.kms.service.Serde;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+class InMemoryEdekSerde implements Serde<InMemoryEdek> {
+
+    @Override
+    public InMemoryEdek deserialize(@NonNull ByteBuffer buffer) {
+        short numAuthBits = Serde.getUnsignedByte(buffer);
+        var ivLength = Serde.getUnsignedByte(buffer);
+        var iv = new byte[ivLength];
+        buffer.get(iv);
+        int edekLength = buffer.limit() - buffer.position();
+        var edek = new byte[edekLength];
+        buffer.get(edek);
+        return new InMemoryEdek(numAuthBits, iv, edek);
+    }
+
+    @Override
+    public int sizeOf(InMemoryEdek inMemoryEdek) {
+        return Byte.BYTES // Auth tag: NIST.SP.800-38D §5.2.1.2 suggests max tag length is 128
```

Comment:
I actually find that harms the readability of the code because these sizes need to agree with what we're writing to the `buffer` in `serialize()`. It's easy see that things agree when we use `Serde.putUnsignedByte()` in one and `Byte.BYTES` in the other. It's harder if we use `Serde.putUnsignedByte()` in one and `SOME_OTHER_CONSTANT` in the other.

We could go to the trouble of modelling the schemas more explicitly by having some primitive `Serdes` for unsigned bytes, byte arrays etc. Then we could use `UNSIGNED_BYTE.serialize(inMemoryEdek.numAuthBits(), buffer)` in the one place and `UNSIGNED_BYTE.sizeOf(inMemoryEdek.numAuthBits())` in the other. But this is test code, so I think it's good enough for now.